client/src/ScanPage/AddressTokensPage.tsx

```diff
@@ -137,52 +156,122 @@ export class AddressTokensPage extends React.Component<
         ))}
       </>
     );
-  }
+  };
 
-  render() {
-    const { error, loading, address, ens, tokens } = this.state;
-    const message = ens ? (
-      <>
-        Hey <span>{ens}!</span> ({address})
-      </>
-    ) : (
-      <>Hey {address}!</>
-    );
+  const getTokens = async () => {
+    try {
+      setState((oldState) => ({ ...oldState, loading: true }));
 
-    return (
-      <main id="site-main" role="main" className="app-content">
-        <div className="container">
-          <div className="content-event years" data-aos="fade-up" data-aos-delay="300">
-            {!error && !loading && <h1>{message}</h1>}
-
-            {error && !loading && (
-              <div className="bk-msg-error">
-                There was an error.
-                <br />
-                Check the address and try again
-              </div>
-            )}
+      const account = delve(match, 'params.account');
+      const addressFromHistory = delve(location, 'state.address');
```

Comment:
What if you have no router state ? Like on this same page I do a refresh? Router state always brings problems unless you handle those edge cases

Reply:
We have this handled in this case in the next line (167). We either use the one from params (no problem here), or as fallback we use the one `addressFromHistory`. This was not implemented by me, using the state. Should we remove it? I didn't want to reimplement it because maybe was used in some case. In the flow I created, it's not used.

<img width="766" alt="Screen Shot 2020-07-27 at 11 08 08" src="https://user-images.githubusercontent.com/8381041/88551632-74b44900-cff9-11ea-927d-3fa7f2e47956.png">